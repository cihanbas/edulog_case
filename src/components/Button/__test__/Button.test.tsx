import { describe, expect, jest, test } from "@jest/globals";
import { fireEvent, render, screen } from "@testing-library/react-native";
import { Button } from "..";

describe("<App />", () => {
  test("Buton Snapshot", () => {
    const onPress = jest.fn();
    const view = render(<Button text="Cihan" onPress={onPress} />);

    expect(view).toMatchSnapshot();
  });
  test("Buton testi ", () => {
    const onPress = jest.fn();
    render(<Button text="Cihan 2" onPress={onPress} />);
    const text = screen.getByTestId("btnText");
    expect(text.props.style.fontFamily).toBe("AvenirLTProBlack");
  });
  test("buton tik", () => {
    const onPress = jest.fn();
    render(<Button text="Cihan 2" onPress={onPress} />);
    const btn = screen.getByTestId("btn");
    fireEvent.press(btn);
    expect(onPress).toHaveBeenCalled();
    expect(onPress).toHaveBeenCalledWith();
  });
});
