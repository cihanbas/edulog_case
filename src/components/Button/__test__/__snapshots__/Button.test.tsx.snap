// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Buton Snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "alignItems": "center",
      },
      {
        "justifyContent": "center",
      },
      {
        "backgroundColor": "#D0810A",
        "borderRadius": 9,
        "marginTop": 8,
        "padding": 10,
      },
    ]
  }
  testID="btn"
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": "AvenirLTProBlack",
        "fontSize": 18,
        "textAlign": "center",
      }
    }
    testID="btnText"
  >
    Cihan
  </Text>
</View>
`;
